aten/src/ATen/native/Linear.cpp

```diff
@@ -75,19 +79,20 @@ Tensor linear(const Tensor& input, const Tensor& weight, const c10::optional<Ten
     return xnnpack::linear(input, weight, *bias);
   }
 #endif
-  if (input.dim() == 2 && bias->defined()) {
+  const auto input_dim = input.dim();
+  if (input_dim == 2 && bias->defined()) {
     // Fused op is marginally faster.
     return at::addmm(*bias, input, weight.t());
   }
-  if (input.dim() == 3 && bias->defined() && !input.is_xla()) {
+  if ((input_dim == 3 || input_dim == 4) && bias->defined() && !input.is_xla()) {
```

Comment:
what is the motivation behind only fusing for 3-dim and 4-dim inputs? any contig inputs should work fine, right? (as can reshape without any perf hit)

Reply:
yea, no motivation, just didnt have real world models to test on. I can update to hit it for all dimensions.

Reply:
in my personal view, the less hidden differences/surprises like this, the better...